pkgs/development/coq-modules/QuickChick/default.nix

```diff
@@ -30,8 +33,42 @@ mkCoqDerivation {
   release."20170512".sha256 = "033ch10i5wmqyw8j6wnr0dlbnibgfpr1vr0c07q3yj6h23xkmqpg";
   releaseRev = v: "v${v}";
 
-  preConfigure = lib.optionalString recent
-    "substituteInPlace Makefile --replace quickChickTool.byte quickChickTool.native";
+  preConfigure =
+    let
+      ocamlBuildFileSwitch =
+        lib.switch [ (if isNull version then defaultVersion else version) ] [
+          { cases = [ "1.6.1" ];    out = { doPatch = true; file = "plugin/quickChick.mlg.cppo"; }; }
+          { cases = [ "1.6.0" ];    out = { doPatch = true; file = "plugin/quickChick.mlg.cppo"; }; }
+          { cases = [ "1.5.1" ];    out = { doPatch = true; file = "plugin/quickChick.mlg"; }; }
+          { cases = [ "1.5.0" ];    out = { doPatch = true; file = "src/quickChick.mlg"; }; }
+          { cases = [ "1.4.0" ];    out = { doPatch = true; file = "src/quickChick.mlg"; }; }
+          { cases = [ "1.3.2" ];    out = { doPatch = true; file = "src/quickChick.mlg"; }; }
+          { cases = [ "1.2.1" ];    out = { doPatch = true; file = "src/quickChick.mlg"; }; }
+          { cases = [ "1.1.0" ];    out = { doPatch = true; file = "src/quickChick.mlg"; }; }
+          { cases = [ "1.0.0" ];    out = { doPatch = false; }; }
+          { cases = [ "" ];         out = { doPatch = false; }; }
+          { cases = [ "20190311" ]; out = { doPatch = false; }; }
+          { cases = [ "20171102" ]; out = { doPatch = false; }; }
+          { cases = [ "20170512" ]; out = { doPatch = false; }; }
+        ] { doPatch = true; file = "plugin/quickChick.mlg.cppo"; };
+
+      ocamlBuildFile =
+        if isNull ocamlBuildFilePatch
+        then ocamlBuildFileSwitch
+        else ocamlBuildFilePatch;
+
+      oldBuildLine = "; ocamlbuild ";
+      newBuildLine = "; ocamlbuild -use-ocamlfind ";
+
+      # Patch ocamlbuild call in QuickChick so it uses ocamlfind to find libraries in nix.
+      patchOcamlBuild =
+        "substituteInPlace ${ocamlBuildFile.file} --replace \"${oldBuildLine}\" \"${newBuildLine}\"";
+
+      patchNative =
+        lib.optionalString recent
+          "substituteInPlace Makefile --replace quickChickTool.byte quickChickTool.native";
+    in
+      lib.concatStringsSep "\n" [ patchNative patchOcamlBuild ];
 
   mlPlugin = true;
   extraNativeBuildInputs = lib.optional recent coq.ocamlPackages.ocamlbuild;
```

Comment:
A simpler fix is to change this line to:
`extraNativeBuildInputs = lib.optionals recent [ coq.ocaml coq.ocamlPackages.ocamlbuild ];`

Reply:
This seems to work in the test repo you linked, I can push a PR to your repo.

Reply:
Thanks for finding this and pinging me @vbgl :pray: I saw this on some OCaml package as well and think I got it fixed